pkg/nodeagent/files/files.go

```diff
@@ -141,6 +141,14 @@ func Move(fs afero.Afero, source, destination string) error {
 func moveCrossDevice(fs afero.Afero, source, destination string) error {
 	tmpFilePath := destination + ".tmp"
 
+	if tmpFile, err := fs.Stat(tmpFilePath); err == nil && tmpFile.Mode().IsRegular() {
```

Comment:
`Copy` uses a temporary directory as destination ([ref](https://github.com/gardener/gardener/blob/3eb256e9a8895d512ee890cdf316cbb28e1dd70d/pkg/nodeagent/files/files.go#L45-L50)) which is deleted afterwards.
However, it is now removed with `RemoveAll`. Otherwise, it cannot be deleted if it is not empty e.g. when the file cannot be moved its the real destination.